packages/react/src/components/ChoiceGroup/ChoiceGroup.base.tsx

```diff
@@ -127,9 +132,20 @@ export const ChoiceGroupBase: React.FunctionComponent<IChoiceGroupProps> = React
     [onChange, options, setKeyChecked],
   );
 
+  const onRadioFocus = React.useCallback(
+    (evt: React.FocusEvent<HTMLElement>) => {
+      // Override focus behavior when the focus comes from oustide the DOM subtree of the radio group.
+      // Handles scenarios like this bug: https://github.com/microsoft/fluentui/issues/20173
+      if (evt.relatedTarget && !elementContains(evt.currentTarget, evt.relatedTarget as HTMLElement)) {
```

Comment:
I decided to leave this as-is since it's only used in this case but I feel this might be better encapsulated into a utility function (in dom utils?) called something like `eventFromSibling(evt)`. It would make this easy to use elsewhere but, more importantly I think it would make this bit of code more readable.

Any thoughts?

Reply:
Maybe split into a function within the file for readability, and then it can be easily moved to utilities if someone else needs it?

Reply:
Good call. Done!